### PR TITLE
Jackson JSON factory feature support

### PR DESCRIPTION
**Pull request by Praytic**
Support for issue #4768 (Support Jackson Configuration for JsonFactory).

### CODE REVIEW

**Comment by jameskleeh** [comment]
Please merge 3.1.x into your branch

File: runtime/src/main/java/io/micronaut/jackson/ObjectMapperFactory.java [hunk 1]
```diff
@@ -80,6 +82,21 @@
...
+    /**
+     * Builds default Factory {@link JsonFactory} using properties from {@link JacksonConfiguration}.
+     *
+     * @param jacksonConfiguration The configuration
+     * @return The {@link JsonFactory}
+     */
+    @Requires(beans = JacksonConfiguration.class)
+    @Singleton
+    @BootstrapContextCompatible
+    public JsonFactory jsonFactory(JacksonConfiguration jacksonConfiguration) {
```
Comment (graemerocher) [inline runtime/src/main/java/io/micronaut/jackson/ObjectMapperFactory.java:94]: does the `JsonFactory` not need to be wired into the `ObjectMapper`?

Comment (jameskleeh) [reply runtime/src/main/java/io/micronaut/jackson/ObjectMapperFactory.java:94]: It already is https://github.com/micronaut-projects/micronaut-core/blob/3.0.x/runtime/src/main/java/io/micronaut/jackson/ObjectMapperFactory.java#L95

Comment (graemerocher) [reply runtime/src/main/java/io/micronaut/jackson/ObjectMapperFactory.java:94]: ok

**Comment by Praytic** [comment]
@jameskleeh pipelines are waiting for approval after I fixed the target branch.

**Comment by Praytic** [comment]
What should I do if unrelated tests are failing? Locally and in CI different tests are failing and neither is related to my changes.

**Comment by ilopmar** [comment]
@Praytic you can try merging `3.1.x` into your branch because at this moment all tests in the branch pass: https://github.com/micronaut-projects/micronaut-core/actions/runs/1212751139

**Comment by jameskleeh** [comment]
@Praytic Please add a test where you configure the factory and then inject the factory and assert the settings have taken effect

**Comment by jameskleeh** [comment]
Thanks for the PR!

**Pull request merged**